src/main/scala/z3/scala/Z3Context.scala

```diff
@@ -1024,22 +1029,8 @@ sealed class Z3Context(val config: Map[String, String]) {
     res
   }
 
-  // XXX: this is a HUGE hack to get around the fact that the Z3 api doesn't
-  //      provide a handle to the absolute value function.
   def getAbsFuncDecl(): Z3FuncDecl = {
-    val ast = parseSMTLIB2String("""
```

Comment:
For what it's worth, given the `mkAbs` helper you added below it should be possible to make this marginally less hacky by avoiding the parsing. I had to do something similar to get a handle to the built-in if-then-else function here: https://github.com/epfl-lara/inox/blob/2f402f9bf9541cc061558e525fdd53efa5f86a87/src/main/scala/inox/solvers/z3/Z3Native.scala#L478